File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        first= strs[0]
        n = len(first)
        prefix = []
        for i in range(n):
            char = first[i]
            for s in strs[1:]:
                if i >= len(s) or s[i] != char:
                    return "".join(prefix)
            prefix.append(char)
        return "".join(prefix)
                    




       

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) 
    {  if ( strs == null || strs.length == 0)
        {
        return "";
        }
        String shortest = strs[0];
        for( String s : strs)
        {
            if ( s.length() < shortest.length())
            {
                shortest = s;
            }
           
        }
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i< shortest.length(); i++)
        {
         for(String s :strs)
         {
            if(shortest.charAt(i) != s.charAt(i))
            {
               return sb.toString();
            }           
         }
         sb.append(shortest.charAt(i));  
        } 
        return sb.toString();
    }
}